client/post-editor/media-modal/detail/detail-item.jsx

```diff
@@ -84,6 +72,63 @@ class EditorMediaModalDetailItem extends Component {
 		);
 	}
 
+	handleOnRestoreClick = () => {
+		const { site, item, onRestore } = this.props;
+		onRestore( site && site.ID, item );
+	};
+
+	renderRestoreButton() {
+		const {
+			item,
+			translate
+		} = this.props;
+
+		//do a simple guid vs url check
+		const guidParts = url.parse( item.guid );
+		const URLParts = url.parse( item.URL );
+
+		if ( guidParts.pathname === URLParts.pathname ) {
```

Comment:
I'm not sure of this since it's a temporal approach. In theory, we could update this code once the new media endpoints are available.